storage/innobase/log/log0files_finder.cc

```diff
@@ -281,7 +288,12 @@ Log_files_find_result log_files_find_and_analyze(
     /* The file can be opened - checked by
     the log_collect_existing_files() */
     if (!file_handle.is_open()) {
+#ifdef XTRABACKUP
+      missing_files.insert(file_index++);
```

Comment:
Is it possible to use file.m_id instead of new file_index?

Reply:
We are using an iterator here. If we try to remove this in place the for loop will jump one file. If this is a matter of keeping the missing_files set and just using the m_id, we should check if the files will not get reordered after a removal. I will check

Reply:
Right, the final order is also important. The invariants before and after removal should remain same.

Reply:
Done

Reply:
Thanks for implementing. It looks good now